File: 0415-add-strings/0415-add-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    const l1 = num1.length;
    const l2 = num2.length;

    let i = l1 - 1;
    let j = l2 - 1;

    let carry = 0;

    let str = "";

    while(i >= 0 && j >= 0){
        let n1 = Number(num1[i--]);
        let n2 = Number(num2[j--]);

        let sum = n1 + n2 + carry;
        carry = Math.floor(sum / 10);

        str = String(sum % 10) + str;

    }
    while(i >= 0){
        let n1 = Number(num1[i--]);

        let sum = n1 + carry;
        carry = Math.floor(sum / 10);

        str = String(sum % 10) + str;

    }
    while(j >= 0){
        let n2 = Number(num2[j--]);

        let sum = n2 + carry;
        carry = Math.floor(sum / 10);

        str = String(sum % 10) + str;

    }

    if(carry != 0) str = String(carry) + str;
    return str;
};

File: 0415-add-strings/0415-add-strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        int l1 = num1.length();
        int l2 = num2.length();

        int len = (l1 > l2)? l1 : l2;

        StringBuilder sum = new StringBuilder("");
        int carry = 0;

        int i = l1 - 1;
        int j = l2 - 1;


        while(i >= 0 && j >= 0){
            int n1 = (int)(num1.charAt(i--) - '0');
            int n2 = (int)(num2.charAt(j--) - '0');
            
            int addition = n1 + n2 + carry;
            carry = addition / 10;


            sum.insert(0, (char)((addition % 10) + '0'));

        }


        while(i >= 0){
            int n1 = (int)(num1.charAt(i--) - '0') + carry;            
            carry = n1 / 10;
            sum.insert(0, (char)((n1 % 10) + '0'));
        }
        while(j >= 0){
            int n2 = (int)(num2.charAt(j--) - '0') + carry;            
            carry = n2 / 10;
            sum.insert(0, (char)((n2 % 10) + '0'));
        }
        if(carry != 0) sum.insert(0, (char)(carry + '0'));
        return sum.toString();
    }
}